### PR TITLE
Formatter class

### LINKED ISSUE #13: Formatting utility class
A formatting utility class supporting Discords formatting syntax.
Example:
```ts
const msg = new Formatter()
  .bold(italic('Code block'))
  .escape('```c')
  .append('printf("Hello world!");')
  .escape('```')
  .codeBlock('ts', 'printf("Hello world!");')
  .build()
```

Note the option to call `bold` on the formatter and `italic` standalone. All functions should be callable in both forms to allow combinations

### LINKED ISSUE #15: Proper formatting in code
Use new class from #14 to write proper formatting in commands.

### PR DESCRIPTION
**Pull request by DrunkenToast**
For #13

### CODE REVIEW

**Review by xunafay — CHANGES_REQUESTED** [review]
combine `.text()` and `.append()` into one function using the optional param. Propagate the optional new line param down in the other functions

File: src/formatter.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  return `\n\`\`\`${lang}\n${val}\n\`\`\``;
+}
+
+export function command(val: string): string {
+  return `\`${val}\``;
+}
+
+export function escape(val: string): string {
+  //https://regex101.com/r/o5VvwW/3
+  return val.replace(/`/g, '\\`').replace(/\*/g, '\\*');
```
Comment (xunafay) [inline src/formatter.ts]: also escape strike through, spoiler and underline
slightly more complicated as `__test__test2__` has to become `\__test\__test2__` aka no pairs may exist

**Review by xunafay — DISMISSED** [review]

File: src/formatter.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+export function escape(val: string): string {
+  //https://regex101.com/r/o5VvwW/3
+  return val
+    .replace(/`/g, '\\`') //backticks for codeblocks and commands
+    .replace(/\*/g, '\\*') //italic and bold
+    .replace(/\|\|/g, '\\||') //spoiler
+    .replace(/~~/g, '\\~~') //strikethrough
+    .replace(/_/g, '\\_');  //underline
+
```
Comment (xunafay) [inline src/formatter.ts]: remove empty newline

File: src/formatter.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+export function codeblock(lang: string, val: string): string {
+  return `\n\`\`\`${lang}\n${val}\n\`\`\``;
+}
+
+export function command(val: string): string {
+  return `\`${val}\``;
+}
+
+export function escape(val: string): string {
+  //https://regex101.com/r/o5VvwW/3
```
Comment (xunafay) [inline src/formatter.ts]: remove outdated/unnecessary link

File: src/formatter.ts [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    return this.text(command(val), nl);
+  }
+
+  escape(val: string, nl = false): this {
+    return this.text(escape(val), nl);
+  }
+}
+
+export function bold(val: string): string {
+
```
Comment (xunafay) [inline src/formatter.ts:103]: remove empty newline

File: src/formatter.ts [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+const mySecretMessage: string = new Formatter()
+      .bold('Code block', true) // true for newline, default false
+      .escape('```').append('c')
+      .append('printf("Hello world!");')
+      .escape('```')
+      .codeblock('c', 'printf("Hello world!");')
+      .append('You can also write commands like this:')
+      .escape('`sudo make me sandwhich`').text(' -> ').command('sudo make me sandwhich')
+      .build();
+*/
```
Comment (xunafay) [inline src/formatter.ts]: add as JSDoc to class

File: src/formatter.ts [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+  constructor() {
+    this.body = '';
+  }
+
+  build(): string {
+    return this.body;
+  }
+
+  text(val: string, append = false): this {
+    this.body = append ? `${this.body}${val}\n` : `${this.body}${val}`;
```
Comment (xunafay) [inline src/formatter.ts]: rename append to `nl` in order to keep consistency between functions

**Review by xunafay — CHANGES_REQUESTED** [review]

File: src/discord.ts [hunk 7]
```diff
@@ -4,7 +4,7 @@ import { Client, MessageEmbed } from 'discord.js';
 import * as data from '../cfg/config.json';
 
 // formatter, don't forget .build() for string
-import { Formatter} from './formatter';
+import { Formatter } from './formatter';
```
Comment (xunafay) [inline src/discord.ts:7]: unused import

**Review by xunafay — APPROVED** [review]

**Review by PandaGamersHDNL — APPROVED** [review]

**Pull request merged**